webapp/graphite/render/evaluator.py

```diff
@@ -14,7 +14,7 @@ def evaluateTarget(requestContext, target):
     return result
 
 
-def evaluateTokens(requestContext, tokens, replacements=None):
+def evaluateTokens(requestContext, tokens, replacements=None, pipedArg=None):
```

Comment:
Could you please elaborate how pipedArg is passed here?

Reply:
`pipedArg` is passed when evaluating the rightmost piped function, the value passed in will be the parse tree for the remaining part of the expression.  You can see on line 30 & 31 that the `rightMost` piped call is `pop()`ed off the stack and `tokens` is then passed as `pipedArg` when it's evaluated.  This mechanism is how the piped calls are "converted" back to a regular nested set of function calls.

Reply:
Ah, missed that, sorry. Thanks!

Reply:
No worries, it's definitely the most non-obvious part of the change.  I'll add some comments to explain what's going on there.

Reply:
Comments added